api/src/client/ApiFormationClient.ts

```diff
@@ -1,31 +1,26 @@
-import { FormationClient } from "@domain/types";
+import { FormationClient, HealthCheckMetadata, HealthCheckMethod } from "@domain/types";
 import { LogWriterType } from "@libs/logWriter";
-import { CountriesShortCodes } from "@shared/countries";
-import { getCurrentDateISOString, parseDateWithFormat } from "@shared/dateHelpers";
-import { defaultDateFormat } from "@shared/defaultConstants";
+import { getCurrentDateISOString } from "@shared/dateHelpers";
 import {
-  BusinessSuffix,
-  FormationLegalType,
   FormationSubmitError,
   FormationSubmitResponse,
   GetFilingResponse,
   InputFile,
-  PaymentType,
-  SignerTitle,
-  formationApiDateFormat,
 } from "@shared/formationData";
-import { StateNames, StateShortCodes } from "@shared/states";
 import { UserData } from "@shared/userData";
-import axios from "axios";
+import axios, { AxiosError } from "axios";
 
+import {
+  ApiConfig,
+  ApiErrorResponse,
+  ApiGetFilingResponse,
+  ApiResponse,
+  makePostBody,
+} from "@client/ApiFormationHelpers";
 import { splitErrorField } from "@client/splitErrorField";
-import { getCurrentBusiness } from "@shared/domain-logic/getCurrentBusiness";
-
-type ApiConfig = {
-  account: string;
-  key: string;
-  baseUrl: string;
-};
+import { generateFormationFormData, generateFormationSigner } from "@shared/test";
```

Comment:
boulder - we may want to chat with the rest of the team, if feels weird to use generator functions in our logic

Reply:
I see what you're saying. Pulling test helpers into production code does feel a little icky to me, but I do see the thought behind this.

Chris and I talked about this yesterday, and at the time I didn't really think anything of it. For a couple reasons, I'm more inclined now to say that it would be worth specifically building data for us to test with. A key reason is the formation API does use some conditional logic to validate a payload. If we're just generating stuff randomly we could also get some false negatives in this health check.

Additionally in considering this now after Faraz called it out, I do also think it's a mistake to use a test generator in our production code. This could end up biting us.

My rec here would be to create a separate file that could be used to load in predefined data so that we could test against specific scenarios. This becomes more relevant if we do find the need to check other validation rules, such as business effective date.

Reply:
Made it its own file